nexus/db-queries/src/db/datastore/external_ip.rs

```diff
@@ -147,24 +143,29 @@ impl DataStore {
     ) -> CreateResult<ExternalIp> {
         let ip_id = Uuid::new_v4();
 
-        let pool_id = match params.pool {
+        // TODO: NameOrId resolution should happen a level higher, in the nexus function
+        let (.., authz_pool, pool) = match params.pool {
```

Comment:
Maybe we could just create a separate issue about this. I think there's a few places in the networking APIs around BGP that have the same thing.

Reply:
I don't think you were planning on it, but let's save the fix for this for another PR.